lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -55,10 +55,16 @@ defmodule Mix.Tasks.Escript.Install do
     :ok
   end
 
-  def after_install(dst, _previous) do
-    File.chmod!(dst, @escript_file_mode)
-    write_bat!(dst <> ".bat", :os.type)
-    check_discoverability(dst)
+  def after_install(dst, binary, _previous) do
+    if escript?(binary) do
+      File.chmod!(dst, @escript_file_mode)
+      write_bat!(dst <> ".bat", :os.type)
+      check_discoverability(dst)
+      :ok
+    else
+      File.rm(dst)
```

Comment:
Maybe we should use `File.rm_rf`? Who knows what may happen to the file before we reach this line. :bowtie: